internal/util/pkgutil/pkgutil.go

```diff
@@ -15,25 +15,67 @@
 package pkgutil
 
 import (
-	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 
 	kptfilev1alpha2 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
 	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
 	"sigs.k8s.io/kustomize/kyaml/copyutil"
 )
 
-type RemoteSubPkgInfo struct {
-	PackagePath          string
-	DeclaringKptfilePath string
+// FindAllDirectSubpackages returns the paths to all immediate subpackages of the
+// provided path. It finds both remote and local subpackages, but it will not
+// find any nested packages. The path to the root package is not returned.
+func FindAllDirectSubpackages(path string) ([]string, error) {
+	excludedDirs := make(map[string]bool)
+	err := filepath.Walk(path, func(p string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// don't copy the .git dir
+		if p != path {
+			rel := strings.TrimPrefix(p, path)
+			if copyutil.IsDotGitFolder(rel) {
+				return nil
+			}
+		}
+
+		for dir := range excludedDirs {
```

Comment:
What are excluded dirs ? Can you add more comments as to what this method is doing, explaining the variables.

Reply:
This was unfortunate naming of the variables. I have updated the code and added some more comments.